core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java

```
@@ -0,0 +1,506 @@
+      absolute = '/' + path;
+    }
+    String[] subPaths = absolute.split("/");
+    ITable table = ROOT;
+    for (String p : subPaths) {
+      table = table.getSubTable(p);
+    }
+    return table;
+  }
+
+  /**
+   * Gets the type of the value for the given key, or null if there's no relevant value.
+   */
+  private Class getExistingType(String key) {
+    if (containsKey(key)) {
+      return entries.get(key).getClass();
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Gets the relevant table for the given key e.g. "/foo/bar" -> "/foo".
+   */
+  private MockNetworkTable tableForKey(String key) {
+    if (key.contains("/")) {
+      return (MockNetworkTable) getSubTable(key.substring(0, key.lastIndexOf('/')));
+    } else {
+      return this;
+    }
+  }
+
+  /**
+   * Gets the name of the given key, e.g. "/foo/bar" -> "bar".
+   */
+  private String realKey(String key) {
+    return key.contains("/") ? key.substring(key.lastIndexOf('/') + 1, key.length()) : key;
+  }
+
+  @Override
+  public boolean containsKey(String key) {
+    return entries.containsKey(key);
+  }
+
+  @Override
+  public boolean containsSubTable(String key) {
+    return subTables.containsKey(key);
+  }
+
+  @Override
+  public ITable getSubTable(String key) {
+    if (containsSubTable(key)) {
+      return subTables.get(key);
+    }
+    if (this == ROOT) {
+      String path = key;
+      if (key.charAt(0) == '/') {
+        path = key.substring(1);
+      }
+      return subTables.computeIfAbsent(path, k -> new MockNetworkTable("/" + k));
+    }
+    if (!key.contains("/")) {
+      return subTables.computeIfAbsent(key, k -> new MockNetworkTable(name + "/" + k));
+    }
+    MockNetworkTable t;
+    if (key.charAt(0) == '/') {
+      t = ROOT;
+    } else {
+      t = this;
+    }
+    for (String p : key.split("/")) {
+      t = (MockNetworkTable) t.getSubTable(p);
+    }
+    return t;
+  }
+
+  @Override
+  public Set<String> getKeys(int types) {
+    return null;
```

this should throw `UnsupportedOperationException`